presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java

```diff
@@ -629,7 +630,12 @@ public HiveSessionProperties(HiveClientConfig hiveClientConfig, OrcFileWriterCon
                         hiveClientConfig.getQuickStatsBackgroundBuildTimeout(),
                         false,
                         value -> Duration.valueOf((String) value),
-                        Duration::toString));
+                        Duration::toString),
+                dataSizeSessionProperty(
+                        AFFINITY_SCHEDULING_FILE_SECTION_SIZE,
+                        "Size of file section for affinity scheduling. Each section of a give size may be assigned an independent affinity preference.",
```

Comment:
of a give size --> of a given size
Also, is that true? If there are five different sizes there can be five different affinity preferences? How does that work?

Reply:
It does sound confusing. Maybe you can help me with a concise and more intuitive description.

Let me try to elaborate more on what is going on here.

Currently preferred nodes for affinity scheduling are assigned based on the file path. However it is hard to guarantee the files being similar is size and relatively small. Hence we are running into a skew.

Here's a real example from one of our production systems:

One of the input tables consist of multiple partitions. Each partition consist of a single file (~30GB in size). Queries usually read a single partition at a time, and multiple queries are submitted hitting a single partition at the same time.

Now since there's only a single file per partition, the partition is always read by a single "preferred" node creating a major bottleneck.

The idea is to increase granularity by assigning different preferred nodes to different sections of a file (256MB by default). So a single file can be cached and served by more than a single node.

Reply:
Thanks. I suspect right here you should simply delete "Each section of a give size may be assigned an independent affinity preference." "Size of file section for affinity scheduling" is all that's needed to describe this property. A detailed description of affinity scheduling can be added elsewhere in the prose docs. It's perhaps worthy of an entire page of its own.